File: Spiky_Client/Source/Spiky_Client/Private/Handlers/Logics/MainMenu.cpp
// Copyright (c) 2017, Vadim Petrov - MIT License

#include "Spiky_Client.h"
#include "MainMenu.h"
#include "Protobufs/MainMenuModels.pb.h"
#include "Descriptors.h"
#include "Chats.h"

void UMainMenu::Handler(MainMenu mainMenu)
{
	if (mainMenu.GetReflection()->HasField(mainMenu, Descriptors::chat_mm))
	{
		UChats * chats = NewObject<UChats>(UChats::StaticClass());
		chats->Handler(mainMenu.chat(), "mainmenu");
	}
}

File: Spiky_Server/src/main/java/com/spiky/server/tcp/logics/MainMenu.java
/*
 * Copyright (c) 2017, Vadim Petrov - MIT License
 */

package com.spiky.server.tcp.logics;

import com.spiky.server.protomodels.MainMenuModels;
import io.netty.channel.ChannelHandlerContext;

import static com.spiky.server.utils.Descriptors.chat_mm;

public class MainMenu {
    public MainMenu(ChannelHandlerContext ctx, MainMenuModels.MainMenu mainMenu) {
        if(mainMenu.hasField(chat_mm))
        {
            new MainMenuChat(ctx, mainMenu.getChat());
        }
    }
}
